tests/db_functions/test_datetime.py

```diff
@@ -198,6 +200,37 @@ def test_extract_func(self):
         self.assertEqual(DTModel.objects.filter(start_date__month=Extract('start_date', 'month')).count(), 2)
         self.assertEqual(DTModel.objects.filter(start_time__hour=Extract('start_time', 'hour')).count(), 2)
 
+    @skipUnlessDBFeature('has_native_duration_field')
+    def test_extract_duration(self):
+        start_datetime = microsecond_support(datetime(2015, 6, 15, 14, 30, 50, 321))
+        end_datetime = microsecond_support(datetime(2016, 6, 15, 14, 10, 50, 123))
+        if settings.USE_TZ:
+            start_datetime = timezone.make_aware(start_datetime, is_dst=False)
+            end_datetime = timezone.make_aware(end_datetime, is_dst=False)
+        self.create_model(start_datetime, end_datetime)
+        self.create_model(end_datetime, start_datetime)
+
+        self.assertQuerysetEqual(
+            DTModel.objects.annotate(extracted=Extract('duration', 'epoch')).order_by('start_datetime'),
```

Comment:
You cannot use a `EXTRACT` with `EPOCH` on Oracle (see [doc](https://docs.oracle.com/cd/B28359_01/server.111/b28286/functions052.htm#SQLRF00639)). Please use another lookup.